tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -11,130 +11,201 @@
-            log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");
-            log_info(test_cur_vmrss << "\t\t" << test_cur_vmhwm << "\t\t" << test_cur_vmsize << "\t\t" << test_cur_vmpeak);
-            seconds++;
-        }
-    } while (t_diff.count() < 5);
 
-    log_debug_ref_record_for_test("infer_request_inference");
+    auto test_pipeline = [&]{
+        MemCheckPipeline memCheckPipeline;
 
-    if ((!Environment::Instance().getCollectResultsOnly()) && (test_cur_vmrss > ref_vmrss))
-        return TestResult(TestStatus::TEST_FAILED,
-                          "Test failed: RSS virtual memory consumption became greater than reference.\n"
-                          "Reference RSS memory consumption: " + std::to_string(ref_vmrss) + " KB.\n" +
-                          "Current RSS memory consumption: " + std::to_string(test_cur_vmrss) + " KB.\n");
+        log_info("Memory consumption before run:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
 
-    if ((!Environment::Instance().getCollectResultsOnly()) && (test_cur_vmhwm > ref_vmhwm))
-        return TestResult(TestStatus::TEST_FAILED,
-                          "Test failed: HWM (peak of RSS) virtual memory consumption is greater than reference.\n"
-                          "Reference HWM of memory consumption: " + std::to_string(ref_vmhwm) + " KB.\n" +
-                          "Current HWM of memory consumption: " + std::to_string(test_cur_vmhwm) + " KB.\n");
+        Core ie;
+        log_info("Memory consumption after Core creation:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
 
-    return TestResult(TestStatus::TEST_OK, "");
+        ie.GetVersions(target_device);
+        log_info("Memory consumption after GetCPPPluginByName (via GetVersions):");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
+
+        CNNNetwork cnnNetwork = ie.ReadNetwork(model_path);
+        log_info("Memory consumption after ReadNetwork:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
+
+        ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, target_device);
+        log_info("Memory consumption after LoadNetwork:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
+
+        InferRequest inferRequest = exeNetwork.CreateInferRequest();
+        log_info("Memory consumption after CreateInferRequest:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
+
+        inferRequest.Infer();
+        OutputsDataMap output_info(cnnNetwork.getOutputsInfo());
+        for (auto &output : output_info)
+            Blob::Ptr outputBlob = inferRequest.GetBlob(output.first);
+        log_info("Memory consumption after Inference:");
+        memCheckPipeline.do_measures();
+        memCheckPipeline.print_measures();
```

Comment:
I created `do_measures_n_print_w_msg` func before, but it was a bad solution from your point of view. I may propose the next: `do_measures_n_print()` func which will be called as `log_info("Some message: " << do_measures_n_print())`

Reply:
I created `do_measures_n_print_w_msg` func before, but it was a bad solution from your point of view. I may propose the next: `log_info(do_measures << "Some message: \n" << get_measures_as_str())`, but I don't see a real advantage of it

Reply:
I added `print_actual_measures()` to combine `do_measures()` and `print_measures()`. Because `print_measures()` calls `log_info` and return nothing, it's hard to combine with message like `log_info("My message" << print_actual_measures())`

Reply:
Discussed offline. Refactored to new MemCheckPipeline API